src/backend/executor/nodeAgg.c

```diff
@@ -1950,6 +1950,21 @@ ExecInitAgg(Agg *node, EState *estate, int eflags)
 	 * initialize child nodes
 	 */
 	outerPlan = outerPlan(node);
+	if (outerPlan->type == T_ExternalScan) {
+		/*
+		 * Hack to indicate to PXF when there is an external scan
+		 */
+		if (aggstate->aggs && list_length(aggstate->aggs) == 1) {
+			foreach(l, aggstate->aggs) {
+				AggrefExprState *aggrefstate = (AggrefExprState *) lfirst(l);
```

Comment:
just by changing `lfirst(l)` to `linitial(aggstate->aggs)`, you can remove `foreach` loop.